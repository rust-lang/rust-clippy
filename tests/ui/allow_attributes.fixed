// run-rustfix
#![allow(unused)]
#![warn(clippy::allow_attributes)]
#![feature(lint_reasons)]

fn main() {}

// Using clippy::needless_borrow just as a placeholder, it isn't relevant.

// Should lint
#[expect(dead_code)]
struct T1;

struct T2; // Should not lint
#[deny(clippy::needless_borrow)] // Should not lint
struct T3;
#[warn(clippy::needless_borrow)] // Should not lint
struct T4;
// `panic = "unwind"` should always be true
#[cfg_attr(panic = "unwind", expect(dead_code))]
struct CfgT;

fn ignore_inner_attr() {
    #![allow(unused)] // Should not lint
}
