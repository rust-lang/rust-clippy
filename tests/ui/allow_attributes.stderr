error: #[allow] attribute found
  --> $DIR/allow_attributes.rs:11:3
   |
LL | #[allow(dead_code)]
   |   ^^^^^ help: replace it with: `expect`
   |
   = note: `-D clippy::allow-attributes` implied by `-D warnings`

error: #[allow] attribute found
  --> $DIR/allow_attributes.rs:20:30
   |
LL | #[cfg_attr(panic = "unwind", allow(dead_code))]
   |                              ^^^^^ help: replace it with: `expect`

error: aborting due to 2 previous errors

